src/hub.js

```diff
@@ -802,51 +802,83 @@ document.addEventListener("DOMContentLoaded", async () => {
+
   if (oauthFlowPermsToken) {
     Cookies.remove(OAUTH_FLOW_PERMS_TOKEN_KEY);
-    joinPayload.perms_token = oauthFlowPermsToken;
   }
 
-  const { token } = store.state.credentials;
-  if (token) {
-    console.log(`Logged into account ${jwtDecode(token).sub}`);
-    joinPayload.auth_token = token;
-  }
+  const createHubChannelParams = permsToken => {
+    const params = {
+      profile: store.state.profile,
+      push_subscription_endpoint: pushSubscriptionEndpoint,
+      auth_token: null,
+      perms_token: null,
+      context: {
+        mobile: isMobile || isMobileVR,
+        hmd: availableVREntryTypes.isInHMD
+      }
+    };
 
-  const hubPhxChannel = socket.channel(`hub:${hubId}`, joinPayload);
+    if (permsToken) {
+      params.perms_token = hubChannel.oauthFlowPermsToken;
+    }
+
+    const { token } = store.state.credentials;
+    if (token) {
+      console.log(`Logged into account ${jwtDecode(token).sub}`);
+      params.auth_token = token;
+    }
+
+    return params;
+  };
+
+  const migrateToNewReticulumServer = async deployNotification => {
+    // On Reticulum deploys, reconnect after a random delay until pool + version match deployed version/pool
+    console.log(`Reticulum deploy detected v${deployNotification.ret_version} on ${deployNotification.ret_pool}`);
+    clearInterval(retDeployReconnectInterval);
+
+    setTimeout(() => {
+      const tryReconnect = async () => {
+        invalidateReticulumMeta();
+        const reticulumMeta = await getReticulumMeta();
+
+        if (
+          reticulumMeta.pool === deployNotification.ret_pool &&
+          reticulumMeta.version === deployNotification.ret_version
+        ) {
+          console.log("Reticulum reconnecting.");
+          clearInterval(retDeployReconnectInterval);
+          const oldSocket = retPhxChannel.socket;
+          const socket = connectToReticulum(isDebug, oldSocket.params());
+          retPhxChannel = await migrateChannelToSocket(retPhxChannel, socket);
+          await hubChannel.migrateToSocket(socket, createHubChannelParams(oauthFlowPermsToken));
+          authChannel.setSocket(socket);
+          linkChannel.setSocket(socket);
+
+          // Disconnect old socket after a delay to ensure this user is always registered in presence.
+          setTimeout(() => {
+            console.log("Reconnection complete. Disconnecting old reticulum socket.");
+            oldSocket.teardown();
+          }, 10000);
+        }
+      };
+
+      retDeployReconnectInterval = setInterval(tryReconnect, 5000);
+      tryReconnect();
+    }, Math.floor(Math.random() * retReconnectMaxDelayMs));
+  };
+
+  retPhxChannel.on("notice", async data => {
+    // On Reticulum deploys, reconnect after a random delay until pool + version match deployed version/pool
+    if (data.event === "ret-deploy") {
+      await migrateToNewReticulumServer(data);
+    }
+  });
+
+  const hubPhxChannel = socket.channel(`hub:${hubId}`, createHubChannelParams());
```

Comment:
nice catch, I had it backwards -- I just wanted to include it in the main flow, but omit it from the reconnect flow, since my understanding is it is a one-time use token

Reply:
Hmm, you're right, it is a one-time use token. Now that I think about it, discord users with unverified emails will not be able to re-join successfully after a reconnect. I'm not sure if they will be properly redirected to oauth either, so it's something we should test. This may just be a good reason to remove that flow, as you suggested, and just inform users that they need accounts with verified emails.